app/views/spree/order_mailer/_payment.html.haml

```diff
@@ -3,12 +3,15 @@
     - if @order.paid?
       = t :email_payment_paid
     - else
-      = t :email_payment_not_paid
+      - if @order.outstanding_balance?
+        = @order.outstanding_balance < 0 ? t(:credit_owed) : t(:balance_due)
+        \:
+        %strong= @order.display_outstanding_balance
+      - else
+        = t :email_payment_not_paid
   %strong
     = t :email_payment_summary
-%h4
-  = t :email_payment_method
-  %strong=  last_payment_method(@order)&.name
-%p
-  %em= last_payment_method(@order)&.description
-%p &nbsp;
+- if @order.payments.any?
+  = render partial: 'list', locals: { payments: @order.payments }
```

Comment:
the invoices views are calling this partial in order_mailer and then this partial is calling the _list partial in /orders again?
if list is used here in order_mailer it should be here.
I think you can simply move /orders/_list.html.haml  to /order_mailer/_payment_list.html.haml
Is the changed supposed to go into the order emails as well?

Reply:
The `/order_mailer/_payment_list.html.haml` partial is shared currently between the order email and the order invoice. And what I'm doing here is replicating the table from `app/views/spree/admin/payments/_list.html.haml` in the invoice and email.

I assume that the email should also be changed as the same problem exists there, in credit owing and balance due situations the info is incorrect.

What is fiddly is that both views are in diff locations, `spree/admin/order` path, the other is at `spree/order_mailer`, which means the latter has access to helper methods like pretty_time (that I figured out come from the `Spree::Admin::BaseHelper`) and the former does not, so I put the partial in `spree/admin/order` but now I see it isn't found from the mailer 😅 .

Options that spring to mind:
- Should I move this partial to `app/views/spree/admin/shared`?
- I'm pretty sure we don't want the Spree based pretty_time method in more places, is there an alternate time formatter/technique we prefer that doesn't depend on Spree?
- why is the order_mailer outside the admin dir anyway? Isn't it part of the orders admin form?
- leave the order_mailer alone and make this much clearer with a new invoice view

Reply:
Update - I'm trying out the shared partial idea - seems promising